files/report.rb

```diff
@@ -116,22 +122,20 @@ def metrics_to_hash(report)
       report_status["skipped"] = 0
     end
     # fix for reports that contain no metrics (i.e. failed catalog)
-    if @format > 1 and report.respond_to?(:status) and report.status == "failed"
+    if report.respond_to?(:status) and report.status == "failed"
       report_status["failed"] += 1
     end
     # fix for Puppet non-resource errors (i.e. failed catalog fetches before falling back to cache)
-    report_status["failed"] += report.logs.find_all {|l| l.source =~ /Puppet$/ && l.level.to_s == 'err' }.count
+    report_status["failed"] += report.logs.count {|l| l.source =~ /Puppet$/ && l.level.to_s == 'err' }
 
     return report_status
   end
 
   def m2h metrics
-    h = {}
-    metrics.each do |title, mtype|
-      h[mtype.name] ||= {}
-      mtype.values.each{|m| h[mtype.name].merge!({m[0].to_s => m[2]})}
-    end
-    return h
+    metrics.map do |title, mtype|
+      values = mtype.values.map { |name, _label, value| [name, value] }
```

Comment:
Since writing this I've learned you can use `.to_h` instead of `.map`, making it even easier.